network/main.tf

```diff
@@ -0,0 +1,126 @@
+# --- network/main.tf ---
+
+data "aws_availability_zones" "available" {}
+
+resource "random_integer" "random" {
+    min = 1
+    max = 10
+
+}
+resource "random_shuffle" "az_list" {
+    input = data.aws_availability_zones.available.names
+    result_count = var.max_subnets
+}
+resource "aws_vpc" "atul_vpc" {
+    cidr_block = var.vpc_cidr
+    enable_dns_hostnames = true 
+    enable_dns_support = true
+
+    tags = {
+        Name = "atul_vpc-${random_integer.random.id}"
+    }
+}
+
+resource "aws_subnet" "atul_public_subnet" {
+    count = var.public_sn_count
```

Comment:
Best practices recommend separating count argument from the rest of arguments using newlines